recipe-client-addon/lib/PreferenceExperiments.jsm

```diff
@@ -61,105 +61,21 @@ XPCOMUtils.defineLazyModuleGetter(this, "LogManager", "resource://shield-recipe-
 XPCOMUtils.defineLazyModuleGetter(this, "Preferences", "resource://gre/modules/Preferences.jsm");
 XPCOMUtils.defineLazyModuleGetter(this, "TelemetryEnvironment", "resource://gre/modules/TelemetryEnvironment.jsm");
 
-this.EXPORTED_SYMBOLS = ["PreferenceExperiments"];
-
-const EXPERIMENT_FILE = "shield-preference-experiments.json";
-
-const PREFERENCE_TYPE_MAP = {
-  boolean: Services.prefs.PREF_BOOL,
-  string: Services.prefs.PREF_STRING,
-  integer: Services.prefs.PREF_INT,
-};
-
-const DefaultPreferences = new Preferences({defaultBranch: true});
-
-/**
- * Enum storing Preference modules for each type of preference branch.
- * @enum {Object}
- */
-const PreferenceBranchType = {
-  user: Preferences,
-  default: DefaultPreferences,
-};
+XPCOMUtils.defineLazyModuleGetter(this, "PreferenceManagement", "resource://shield-recipe-client/lib/PreferenceManagement.jsm");
 
-/**
- * Asynchronously load the JSON file that stores experiment status in the profile.
- */
-let storePromise;
-function ensureStorage() {
-  if (storePromise === undefined) {
-    const path = OS.Path.join(OS.Constants.Path.profileDir, EXPERIMENT_FILE);
-    const storage = new JSONFile({path});
-    storePromise = storage.load().then(() => storage);
-  }
-  return storePromise;
-}
+this.EXPORTED_SYMBOLS = ["PreferenceExperiments"];
 
 const log = LogManager.getLogger("preference-experiments");
 
-// List of active preference observers. Cleaned up on shutdown.
-let experimentObservers = new Map();
-CleanupManager.addCleanupHandler(() => PreferenceExperiments.stopAllObservers());
+const PREF_NAMESPACE = "experiments";
 
-this.PreferenceExperiments = {
-  /**
-   * Set the default preference value for active experiments that use the
-   * default preference branch.
-   */
+this.PreferenceExperiments = Object.assign({}, PreferenceManagement(PREF_NAMESPACE), {
```

Comment:
I always use fresh objects when using `Object.assign`, I think as a defensive measure. I'll correct this.